Automattic/wp-calypso
# [Domains] Implement featured domains

This implements featured domains specified in the latest Kracken design. It takes the current `Recommended` and `Best Match` flagged domains and moves them to their own sections above all other results, like so:

<img width="979" alt="wordpress_com" src="https://user-images.githubusercontent.com/4044428/37547874-cfe0356e-2939-11e8-8ee1-f95afd4d1e09.png">

I refactored some related UI code while implementing this feature. If reviewers would prefer, I can spin off some of the refactors to a separate PR to make reviewing this change easier.

# Testing Instructions
1. Spin up your environment locally, which has `domains/kracken-ui` feature flag enabled.
2. Navigate to [`/start/domains`](calypso.localhost:3000/start/domains).
3. Ensure that the featured domains behave as expected. Also verify that it does not deviate significantly from the original design (`p989we-e9`).
